### PR TITLE
Inventory migration

### PR DESCRIPTION
**Pull request by Generalcamo**
Migrates (most) null ItemStack checks to InvTools.isEmpty() and (most) null itemstacks to InvTools.emptyStack().

Requirement for 1.11 (see #1232)

~~Lang and API submodules in branch are broken unfortunately. Should stop using IntelliJ's Version Control.~~

### CODE REVIEW

**Review by liach — DISMISSED** [review]
Push your changes in the language repository to a separate branch in the lang repo first.

**Comment by Generalcamo** [comment]
@liach There are no changes to the lang or API submodules.

File: src/main/java/mods/railcraft/common/gui/containers/ContainerAnvil.java [hunk 1]
```diff
@@ -307,7 +307,7 @@ public void onPickupFromSlot(EntityPlayer playerIn, ItemStack stack) {
                     itemstack.stackSize -= repairContainer.materialCost;
                     repairContainer.inputSlots.setInventorySlotContents(1, itemstack);
                 } else {
-                    repairContainer.inputSlots.setInventorySlotContents(1, null);
+                    repairContainer.inputSlots.setInventorySlotContents(1, InvTools.emptyStack());
                 }
             } else {
                 repairContainer.inputSlots.setInventorySlotContents(1, null);
```
Comment (liach) [inline src/main/java/mods/railcraft/common/gui/containers/ContainerAnvil.java]: hmm

**Comment by CovertJaguar** [comment]
So are we good?

**Comment by liach** [comment]
:+1:

**Comment by Generalcamo** [comment]
@CovertJaguar @liach We good now 👍

**Comment by liach** [comment]
@Generalcamo It is not fixed. See https://github.com/Railcraft/Railcraft/pull/1234/files the first one

**Comment by Generalcamo** [comment]
@liach Finally managed to fix it. This can be merged now.

**Review by CovertJaguar — APPROVED** [review]

**Comment by liach** [comment]
@CovertJaguar Never merge no-ff in the future, please always squash!

**Comment by CovertJaguar** [comment]
No, I will continue to decide on a case by case basis.

**Comment by liach** [comment]
Now I think we should always do full merges. That's the most helpful to the integrity of commits.

**Pull request merged**